src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -2035,6 +2145,633 @@ interface IB
             );
         }
 
+        [Fact]
+        public void Inheritance_12()
+        {
+            var source =
+@"class A
+{
+    public object X { get; }
+    public object Y { get; }
+}
+data class B(object X, object Y) : A
+{
+    public object X { get; }
+    public object Y { get; }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,19): warning CS0108: 'B.X' hides inherited member 'A.X'. Use the new keyword if hiding was intended.
+                //     public object X { get; }
+                Diagnostic(ErrorCode.WRN_NewRequired, "X").WithArguments("B.X", "A.X").WithLocation(8, 19),
+                // (9,19): warning CS0108: 'B.Y' hides inherited member 'A.Y'. Use the new keyword if hiding was intended.
+                //     public object Y { get; }
+                Diagnostic(ErrorCode.WRN_NewRequired, "Y").WithArguments("B.Y", "A.Y").WithLocation(9, 19));
+            var actualMembers = GetProperties(comp, "B").ToTestDisplayStrings();
+            var expectedMembers = new[]
+            {
+                "System.Object B.X { get; }",
+                "System.Object B.Y { get; }",
+            };
+            AssertEx.Equal(expectedMembers, actualMembers);
+        }
+
+        [Fact]
+        public void Inheritance_13()
+        {
+            var source =
+@"data class A(object X, object Y)
+{
+    internal A() { }
+}
+data class B(object X, object Y) : A
+{
+    public object X { get; }
+    public object Y { get; }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,19): warning CS0108: 'B.X' hides inherited member 'A.X'. Use the new keyword if hiding was intended.
+                //     public object X { get; }
+                Diagnostic(ErrorCode.WRN_NewRequired, "X").WithArguments("B.X", "A.X").WithLocation(7, 19),
+                // (8,19): warning CS0108: 'B.Y' hides inherited member 'A.Y'. Use the new keyword if hiding was intended.
+                //     public object Y { get; }
+                Diagnostic(ErrorCode.WRN_NewRequired, "Y").WithArguments("B.Y", "A.Y").WithLocation(8, 19));
+            var actualMembers = GetProperties(comp, "B").ToTestDisplayStrings();
+            var expectedMembers = new[]
+            {
+                "System.Object B.X { get; }",
+                "System.Object B.Y { get; }",
+            };
+            AssertEx.Equal(expectedMembers, actualMembers);
+        }
+
+        [Fact]
+        public void Inheritance_14()
+        {
+            var source =
+@"class A
+{
+    public object P1 { get; }
+    public object P2 { get; }
+    public object P3 { get; }
+    public object P4 { get; }
+}
+class B : A
+{
+    public new int P1 { get; }
+    public new int P2 { get; }
+}
+data class C(object P1, int P2, object P3, int P4) : B
```

Comment:
>object P3 [](start = 32, length = 9)

The spec has changed again. See [https://github.com/dotnet/csharplang/pull/3527](https://github.com/dotnet/csharplang/pull/3527) ... Now it appears P3 *should* be generated.

Reply:
The spec currently allows the base member to differ by signature (or member kind) if the record property would hide the base member:
> Two members are considered matching if they have the same signature or would be considered "hiding" in an inheritance scenario.